### PR TITLE
*: Move remaining docs to version specific directories

### PR DESCRIPTION
**Pull request by morgo**
This PR removes the `dev-guide/` - it was not in the TOC, and had some out of date instructions. They are also duplicated: it is better to link to build instructions in the individual projects.

The following files are moved into `dev/*`:
```
faq/*
releases/*
benchmark/*
tispark/*
adopters.md
architecture.md
key-features.md
community.md
contribute.md
overview.md
report-issue.md 
roadmap.md
support-resources.md
technical-writing-project-ideas.md
```
Links that pointed to these files (or op-guide/sql) have also been cleaned up to go to the new locations.

### CODE REVIEW

**Comment by morgo** [comment]
Note: scripts, templates, TOC.md and media are still in the top level directory. The idea is to move these too, but I am afraid that it might break to do it now... so it can be done in a followup push.

File: dev/how-to/get-started/tidb-binlog.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ category: how-to
 
 This tutorial starts with a simple TiDB-Binlog deployment with a single node of each component (Placement Driver, TiKV Server, TiDB Server, Pump, and Drainer), set up to push data into a MariaDB Server instance.
 
-This tutorial is targeted toward users who have some familiarity with the [TiDB Architecture](https://pingcap.com/docs/architecture/), who may have already set up a TiDB cluster (not mandatory), and who wants to gain hands-on experience with TiDB-Binlog. This tutorial is a good way to "kick the tires" of TiDB-Binlog and to familiarize yourself with the concepts of its architecture.
+This tutorial is targeted toward users who have some familiarity with the [TiDB Architecture](/dev/docs/architecture.md), who may have already set up a TiDB cluster (not mandatory), and who wants to gain hands-on experience with TiDB-Binlog. This tutorial is a good way to "kick the tires" of TiDB-Binlog and to familiarize yourself with the concepts of its architecture.
```
Comment (dcalvin) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
This tutorial is targeted toward users who have some familiarity with the [TiDB Architecture](/dev/architecture.md), who may have already set up a TiDB cluster (not mandatory), and who wants to gain hands-on experience with TiDB-Binlog. This tutorial is a good way to "kick the tires" of TiDB-Binlog and to familiarize yourself with the concepts of its architecture.
```

File: TOC.md [hunk 2]
```diff
@@ -2,379 +2,379 @@
 
 ## Documentation List
 
-+ [Introduction](overview.md)
++ [Introduction](/dev/overview.md)
```
Comment (lilin90) [inline TOC.md]: @morgo Is the `/` character before "dev" added on purpose? 
@YiniXu9506 Do you suggest we add the `/` character before "dev"?

File: dev/faq/tidb.md [hunk 3]
```diff
@@ -545,13 +543,13 @@ When TiDB is executing a SQL statement, the query will be `EXPENSIVE_QUERY` if e
 
 #### How to control or change the execution priority of SQL commits?
 
-TiDB supports changing the priority on a [per-session](/sql/tidb-specific.md#tidb_force_priority), [global](/sql/server-command-option.md#force-priority) or [individual statement basis](/sql/dml.md). Priority has the following meaning:
+TiDB supports changing the priority on a [per-session](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#tidb_force_priority), [global](/dev/reference/configuration/tidb-server/configuration.md#force-priority) or individual statement basis. Priority has the following meaning:
```
Comment (lilin90) [inline dev/faq/tidb.md]:
```suggestion
TiDB supports changing the priority on a [per-session](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#tidb-force-priority), [global](/dev/reference/configuration/tidb-server/configuration.md#force-priority) or individual statement basis. Priority has the following meaning:
```

File: dev/faq/tidb.md [hunk 4]
```diff
@@ -929,22 +927,22 @@ update mysql.tidb set VARIABLE_VALUE="3" where VARIABLE_NAME="tikv_gc_concurrenc
...
 
 ### Statistics collection
 
-See [Introduction to Statistics](/sql/statistics.md).
+See [Introduction to Statistics](/dev/reference/performance/statistics.md).
 
 #### How to optimize `select count(1)`?
 
-The `count(1)` statement counts the total number of rows in a table. Improving the degree of concurrency can significantly improve the speed. To modify the concurrency, refer to the [document](/sql/tidb-specific.md#tidb_distsql_scan_concurrency). But it also depends on the CPU and I/O resources. TiDB accesses TiKV in every query. When the amount of data is small, all MySQL is in memory, and TiDB needs to conduct a network access.
+The `count(1)` statement counts the total number of rows in a table. Improving the degree of concurrency can significantly improve the speed. To modify the concurrency, refer to the [document](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#tidb_distsql_scan_concurrency). But it also depends on the CPU and I/O resources. TiDB accesses TiKV in every query. When the amount of data is small, all MySQL is in memory, and TiDB needs to conduct a network access.
```
Comment (lilin90) [inline dev/faq/tidb.md]:
```suggestion
The `count(1)` statement counts the total number of rows in a table. Improving the degree of concurrency can significantly improve the speed. To modify the concurrency, refer to the [document](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#tidb-distsql-scan-concurrency). But it also depends on the CPU and I/O resources. TiDB accesses TiKV in every query. When the amount of data is small, all MySQL is in memory, and TiDB needs to conduct a network access.
```

File: dev/releases/2.1ga.md [hunk 5]
```diff
@@ -89,17 +90,17 @@ On November 30, 2018, TiDB 2.1 GA is released. See the following updates in this
 
         - Check the TiDB cluster information
 
-    - [Add the `auto_analyze_ratio` system variables to contorl the ratio of Analyze](https://pingcap.com/docs/FAQ/#whats-the-trigger-strategy-for-auto-analyze-in-tidb)
+    - [Add the `auto_analyze_ratio` system variables to contorl the ratio of Analyze](/dev/faq/tidb.md#what-s-the-trigger-strategy-for-auto-analyze-in-tidb)
 
-    - [Add the `tidb_retry_limit` system variable to control the automatic retry times of transactions](https://pingcap.com/docs/sql/tidb-specific/#tidb-retry-limit)
+    - [Add the `tidb_retry_limit` system variable to control the automatic retry times of transactions](/dev/reference/configuration/tidb-server/tidb-specific-variables.md##tidb-retry-limit)
```
Comment (lilin90) [inline dev/releases/2.1ga.md]:
```suggestion
    - [Add the `tidb_retry_limit` system variable to control the automatic retry times of transactions](/dev/reference/configuration/tidb-server/tidb-specific-variables.md#tidb-retry-limit)
```

File: TOC.md [hunk 6]
```diff
@@ -2,379 +2,379 @@
...
 + FAQs
-  - [TiDB FAQs](faq/tidb.md)
-  - [TiDB Lightning FAQs](faq/tidb-lightning.md)
-  - [Upgrade FAQs](faq/upgrades.md)
-+ Support  
-  - [Support Resources](support-resources.md)
-  - [Report an Issue](report-issue.md)
+  - [TiDB FAQs](/dev/faq/tidb.md)
+  - [TiDB Lightning FAQs](/dev/faq/tidb-lightning.md)
+  - [Upgrade FAQs](/dev/faq/upgrades.md)
```
Comment (lilin90) [inline TOC.md]: This file name is changed to `upgrade` to make it consistent with that in docs-cn (https://github.com/pingcap/docs/pull/1209). I think `upgrade` is enough to convey the meaning.

File: TOC.md (hunk 2 above)
Comment (morgo) [reply TOC.md]: It is not required. You are correct: I did this so I could write a script to make sure all links start with `/dev/`. In the next stage, I will remove `/dev/` and use relative links.

File: TOC.md (hunk 6 above)
Comment (morgo) [reply TOC.md]: Fixed.

File: TOC.md (hunk 2 above)
Comment (lilin90) [reply TOC.md]: @morgo Does this PR add the `/` character before all "dev" occurrences in the latest docs version. I'm not sure whether removing `/dev/` works. After removing it, the link would be `how-to/.../...`. This works in `TOC.md` but might not work in .md files in a deep directory. @YiniXu9506 PTAL

Comment (morgo) [reply TOC.md]: @lilin90 yes, I'm not sure if it supports removing `/dev/` yet. If it does, I will address that in a PR immediately following this one. But that is the plan...

Comment (YiniXu9506) [reply TOC.md]: @morgo You can add `/` or remove `/` before the relative path, we have script to make the link works, just make sure the relative path starts from the root of each version. And we don't support removing `/dev/` now, so please keep it, you can remove later once all versions are ready.

File: dev/benchmark/sysbench-v4.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
+---
```
Comment (lilin90) [inline dev/benchmark/sysbench-v4.md]: This file was updated and renamed to `how-to-run-sysbench.md` in https://github.com/pingcap/docs/pull/1213.

File: dev/faq/upgrades.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
+---
```
Comment (lilin90) [inline dev/faq/upgrades.md]: And this file was renamed to `upgrade.md`.

File: TOC.md (hunk 2 above)
Comment (morgo) [reply TOC.md]: OK, then the current PR should be fine.

**Pull request merged**